### PR TITLE
User interface

### PR DESCRIPTION
**Pull request by m-anwar97**
Added user interface and loop turns in main file

### CODE REVIEW

File: bin/main [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+while start == true
+  print '╔═══╦═══╦═══╗ \n ║   ║   ║   ║ \n ╠═══╬═══╬═══╣ \n ║   ║   ║   ║ \n ╠═══╬═══╬═══╣
+  \n ║   ║   ║   ║ \n ╚═══╩═══╩═══╝ \n'
+  puts "#{p1}'s turn : "
+  # p1_position
+  # updates position of player 1 on the board
+  puts "#{p2}'s turn : "
+  # p2_position
+  # updates position of player 2 on the board
+end
```
Comment (tirthajyoti-ghosh) [inline bin/main]: This loop is running infinitely as there is no terminating condition. Please introduce a terminating condition to stop this behavior.

File: bin/main [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+puts "#{p1}, X or O?"
+p1_xo = gets.chomp
+p2_xo = p1_xo == 'X' ? 'O' : 'X'
+puts "#{p1} is #{p1_xo}, #{p2} is #{p2_xo}"
+print '╔═══╦═══╦═══╗ \n ║ 1 ║ 2 ║ 3 ║ \n ╠═══╬═══╬═══╣ \n ║ 4 ║ 5 ║ 6 ║
+\n ╠═══╬═══╬═══╣ \n ║ 7 ║ 8 ║ 9 ║ \n ╚═══╩═══╩═══╝ \n'
+puts 'Instructions : Enter a number from 1 to 9 to use one of the boxes. \n'
+start = true
+while start == true
+  print '╔═══╦═══╦═══╗ \n ║   ║   ║   ║ \n ╠═══╬═══╬═══╣ \n ║   ║   ║   ║ \n ╠═══╬═══╬═══╣
```
Comment (tirthajyoti-ghosh) [inline bin/main]: Since you are using single-quotes (`'`), the escape characters will not work. Use double-quotes (`"`).

**Review by tirthajyoti-ghosh — CHANGES_REQUESTED** [review]
# Changes required 🛑 

## What a remarkable project you have built 💯! This is impressive 🤩👌🧨!

- Great job on creating a decent README at this milestone 👏 !
- You are accepting players' names at the start. This makes your game stand out more. Good job there.

### But in order to get it approved, we need to work on some changes. Let's see...

- While running your game, after player name inputs, the game launches an infinite loop. I have left a comment below about it. Be sure to check it out.

- Great use of inline comments to show the game flow. But you also need to inform the players about their turns in the console. You can convert the comments on lines 21 and 24 to `puts` statements. But you are free to do it however you want.

- You need to display the game board after asking each player to place their tokens. You can display an empty board or tokens placed anywhere you like. You can convert the comments on lines 22 and 25 to `puts` statements. But you are free to do it however you want.

- Your game should be able to ask the player where it should place a token (position) after showing available positions. Remember, you just have to mock it. You are not required to implement a game logic.

- Your game should be able to inform a player if the selected move is invalid (without computing what an invalid move is).

- Your game should inform players if the selected move is a winning move (without computing the winning move).

- Your game should inform players if the selected move is a draw move (without computing draw).

- Remember to repeat all actions for the next player's move.

**Review by tirthajyoti-ghosh — CHANGES_REQUESTED** [review]
#### Summary

* [x] `[REQUIRED]` game tells which player turn it is (without computing who the winner is)
* [x] `[REQUIRED]` game asks the player to select from available moves
* [x] `[REQUIRED]` game informs player if the selected move is invalid (without computing what an invalid move is)
* [x] `[REQUIRED]` game displays board after the player move (without computing which spots on the board are taken)
* [x] `[REQUIRED]` game informs player if the selected move is a winning move (without computing the winning move)
* [x] `[REQUIRED]` game informs player if the selected move is a draw move (without computing draw)
* [x] `[REQUIRED]` game repeats all actions for the next player's move

Your project looks absolutely fantastic. Just make these changes and apply for another code review.

Happy coding 👨‍💻 

*Need help? Shoot me a message on Slack - [@Ghosh](https://app.slack.com/client/T47CT8XPG/DRDM6LAQN/user_profile/URB1MHXK6)*
<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)

**Review by peterrobert — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED :recycle:  :stop_sign: 
### HELLO :wave: YOU HAVE DONE SUCH A GREAT JOB :+1: GREAT WORK IMPROVING THE APPLICATION JUST LIKE SUGGESTED BY THE PREVIOUS TSE :+1: 

### JUST SOME FEW MORE IMPROVEMENTS TO BE MADE AND WE SHALL BE GOOD TO GO :smile: 

- [x] The game should display the board after the players move. (without computing which spots on the board are taken).

display board

- [x] The game should be able to tell the player if it's a winning move (without computing the winning move). You don't have to write the logic of this.

- [x]  The game informs the player if the selected move is invalid (without computing what an invalid move is)

### if you have any question concerning the review kindly reach me on slack @peterrobert


### HAPPY CODING :computer:

**Review by peterrobert — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED :recycle: :stop_sign: 
### HELLO :wave: VERY NICE PROGRESS BUT THERE ARE STILL SOME MORE REQUIREMENTS THAT HAVENT BEEN MET YET. KINDLY CHECK BELOW :arrow_down: 

- [x] Your game just outputs that the game is a draw, which is okay but you need also to inform the player if the selected move is a winning move (without computing the winning move)

- [x] `[REQUIRED]` game informs player if the selected move is invalid (without computing what an invalid move is)

### if you have any question kindly reach me on slack @peterrobert 

### HAPPY CODING :computer:

**Comment by krys2fa** [comment]
# Status: Milestone Approved! 🥇 
Congratulations on making all the requested changes and getting this milestone approved. 👍 

I am suggesting some optional changes to make your project even better. They are as follows:

- [ ] [OPTIONAL] The game accepts empty names. You should validate this to ensure your game can be played without any confusion.

- [ ] [OPTIONAL] The game accepts any key as the symbol to be used by the players. You should validate this as well so players can only choose `X` or `O` as the game instructions have prompted them.

You can now merge this branch into the development branch and go on with your next milestone.

💡  Do not forget to submit your project completion form after merging this branch.

All the best team,
Chris

**Pull request merged**